dashboard/app/models/user.rb

```diff
@@ -733,9 +733,16 @@ def primary_script
   end
 
   def needs_to_backfill_user_scripts?
-    user_scripts.empty? && !user_levels.empty?
+    user_scripts.empty? &&
+      !user_levels.empty? &&
+      # Backfill only applies to users created before UserScript model was introduced.
+      created_at < Date.new(2014, 9, 11)
```

Comment:
the created_at check is the least expensive so it would be slightly better to do it first. Also would make it so you don't have a comment in the middle of a multiline conditional ;)